tests/app/main/test_login.py

```diff
@@ -305,14 +409,724 @@ def test_reset_password_form_and_inputs_not_autofillable(
+                data={'email_address': 'email@email.com', 'name': 'valid'}
+            )
+
+            assert_equal(res.status_code, 503)
+
+    @mock.patch('app.main.views.login.send_email')
+    def test_should_call_send_invitation_email_with_correct_params(
+            self, send_email
+    ):
+        with self.app.app_context():
+
+            self.login()
+
+            self.app.config['DM_MANDRILL_API_KEY'] = "API KEY"
+            self.app.config['INVITE_EMAIL_SUBJECT'] = "SUBJECT"
+            self.app.config['INVITE_EMAIL_FROM'] = "EMAIL FROM"
+            self.app.config['INVITE_EMAIL_NAME'] = "EMAIL NAME"
+
+            res = self.client.post(
+                '/suppliers/invite-user',
+                data={'email_address': 'email@email.com', 'name': 'valid'}
+            )
+
+            assert_equal(res.status_code, 302)
+
+            send_email.assert_called_once_with(
+                "email@email.com",
+                mock.ANY,
+                "API KEY",
+                "SUBJECT",
+                "EMAIL FROM",
+                "EMAIL NAME",
+                ["user-invite"]
+            )
+
+    def test_should_be_an_error_for_invalid_token(self):
+        with self.app.app_context():
+            token = "1234"
+            res = self.client.get(
+                '/suppliers/create-user/{}'.format(token)
+            )
+            assert_equal(res.status_code, 400)
+
+    def test_should_be_an_error_for_missing_token(self):
+        with self.app.app_context():
+            res = self.client.get(
+                '/suppliers/create-user'
+            )
+            assert_equal(res.status_code, 404)
+
+    def test_should_be_an_error_for_missing_token_trailing_slash(self):
+        with self.app.app_context():
+            res = self.client.get(
+                '/suppliers/create-user/'
+            )
+            assert_equal(res.status_code, 301)
+            assert_equal(res.location, 'http://localhost/suppliers/create-user')
+
+    @mock.patch('app.main.views.login.data_api_client')
+    def test_should_be_an_error_for_invalid_token_contents(self, data_api_client):
+        with self.app.app_context():
+
+            token = generate_token(
+                {
+                    'this_is_not_expected': 1234
+                },
+                self.app.config['SHARED_EMAIL_KEY'],
+                self.app.config['INVITE_EMAIL_SALT']
+            )
+
+            res = self.client.get(
+                '/suppliers/create-user/{}'.format(token)
+            )
+            assert_equal(res.status_code, 400)
+            assert_equal(data_api_client.get_user.called, False)
+            assert_equal(data_api_client.get_supplier.called, False)
+
+    @mock.patch('app.main.views.login.generate_token')
+    def test_should_be_a_bad_request_if_token_expired(self, generate_token):
+        with self.app.app_context():
+
+            generate_token.side_effect = BadTimeSignature("Too old")
```

Comment:
This should be mocking `decode_invitation_token` here. As it stands the 400 is because the token is invalid.